test/e2e/space_test.go

```diff
@@ -121,18 +120,138 @@ func TestCreateSpace(t *testing.T) {
 	})
 }
 
-func TestUpdateSpace(t *testing.T) {
+func TestSpaceRoles(t *testing.T) {
+
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
 
 	// given
+	s := NewSpace(hostAwait.Namespace, GenerateName("oddity"), "appstudio", WithTargetCluster(memberAwait.ClusterName))
+
+	// when
+	err := hostAwait.CreateWithCleanup(context.TODO(), s)
+
+	// then
+	require.NoError(t, err)
+	// wait until NSTemplateSet has been created and Space is in `Ready` status
+	s, err = hostAwait.WaitForSpace(s.Name,
+		wait.UntilSpaceHasConditions(Provisioned()),
+		wait.UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName),
+	)
+	require.NoError(t, err)
+	// wait until NSTemplateSet has been created and Space is in `Ready` status
+	nsTmplSet, err := memberAwait.WaitForNSTmplSet(s.Name, wait.UntilNSTemplateSetHasConditions(Provisioned()))
+	require.NoError(t, err)
+	tierChecks, err := tiers.NewChecks("appstudio")
+	require.NoError(t, err)
+	tiers.VerifyGivenNsTemplateSet(t, memberAwait, nsTmplSet, tierChecks, tierChecks, tierChecks.GetExpectedTemplateRefs(hostAwait))
+
+	t.Run("add admin role and bindings", func(t *testing.T) {
+		// given
+		adminTierTmpl := NewTierTemplate(t, hostAwait.Namespace, "space-role-admin-123456", "space-role-admin", "appstudio", "123456", []byte(spaceAdminTmpl))
+		err := hostAwait.CreateWithCleanup(context.TODO(), adminTierTmpl)
+		require.NoError(t, err)
+		nsTmplSet.Spec.SpaceRoles = []toolchainv1alpha1.NSTemplateSetSpaceRole{
+			{
+				TemplateRef: adminTierTmpl.Name,
+				Usernames:   []string{"user1", "user2"},
+			},
+		}
+
+		// when
+		err = memberAwait.Client.Update(context.TODO(), nsTmplSet)
```

Comment:
btw, you can again hit the conflict error when updating the NSTemplateSet

Reply:
yes, I saw that too. I've added an `UpdateNSTemplateSet` method in `wait.MemberAwaitility` to fix that. See 89831dd